builtin_bigint.go

```diff
@@ -0,0 +1,369 @@
+package goja
+
+import (
+	"fmt"
+	"hash/maphash"
+	"math"
+	"math/big"
+	"reflect"
+	"strconv"
+	"sync"
+
+	"github.com/dop251/goja/unistring"
+)
+
+type valueBigInt struct {
+	i *big.Int
+}
+
+func (v valueBigInt) ToInteger() int64 {
```

Comment:
`ToInteger()` is effectively a shortcut for `toInt64(ToNumber())`. That means if `ToNumber()` throws, so should `ToInteger()`.

If it's implemented like this it means a lot of places do not throw the "`Cannot convert a BigInt value to a number`" error when they should (like for example `([]).at(1n)`)